### ISSUE TITLE
Apply default memory limit in Mono env

### ISSUE THREAD

**Issue by SteveSandersonMS** [aspnet/AspNetCore#5617]
Recommendation from @kumpera is to set it at 128MB initially, allowing the developer to change that setting for their app if they want.

If we don't do this, and the app exceeds the amount of memory the browser wishes to give, then it fails with an obscure error instead of an OutOfMemoryException.

Note to self: to set env vars for the Mono wasm runtime, call something like the following from JS before the runtime starts.

    MONO.mono_wasm_setenv ("MONO_LOG_MASK", "all");

**Comment by moataz1991** [comment]
im sorry but im stuck !
when we will be able to attach large files in blazor client side ?

**Comment by javiercn** [comment]
See https://github.com/aspnet/AspNetCore/issues/16692 for the GC being too aggressive as a potential outcome of this.

**Comment by legistek** [comment]
I'm sure it's related, but the GC might be too aggressive even leaving aside the memory limits. Even with memory usage in the 20-30MB range I'm getting frequent, blocking GCs:

![image](https://user-images.githubusercontent.com/13384523/67950969-a6a69500-fbb8-11e9-9835-1823080d7a00.png)

Incidentally I tried `GC.TryStartNoGCRegion` during render tree building but I wound up with `NotImplementedException`s.

**Comment by Herdo** [comment]
@legistek Coming from #5466 and this:
![grafik](https://user-images.githubusercontent.com/651191/73206468-42f22080-4143-11ea-8b0b-c7d98274d258.png)

This is the only GC log I see in the console:
![grafik](https://user-images.githubusercontent.com/651191/73206427-31a91400-4143-11ea-9f11-44dfaa94f1f2.png)

**Comment by legistek** [comment]
Hm your issue might not be related then. But overall, Blazor's biggest performance bottleneck definitely seems to be in building very complex DOMs. One would assume AOT would help with that significantly. That said, because the WASM designers inexplicably chose not to give WASM direct access to the DOM, it's always going to require JS interop, which is always going to be an issue.

**Comment by legistek** [comment]
This doesn't seem to have been addressed (at least not fully) in the official release of Blazor WASM. I still get a lot of instances of the below, accompanied by very noticeable freezes in the browser:

![image](https://user-images.githubusercontent.com/13384523/83073468-1c32f080-a036-11ea-9654-562525f3b00b.png)

Granted I am trying to build a complex DOM but nothing that a native JS framework couldn't handle easily. Can't the GC settings be tweaked to allow more RAM usage before browser-freezing GCs take place?

**Comment by Stamo-Gochev** [comment]
What is the expected milestone for this? Will it be reconsidered as wasm is now officially supported?

**Comment by StachuDotNet** [comment]
> @SteveSandersonMS is this still relevant?

Yes - just experienced this in an AOT release build, .NET 6.0.300. Will follow up with more details if useful.

**Comment by SteveSandersonMS** [comment]
This old issue isn't tracking any actual pending action so I'll close it to keep the backlog clearer.

The closest issue related to wasm memory limits is https://github.com/dotnet/aspnetcore/issues/42055
